examples/prometheus/src/main.rs

```diff
@@ -51,6 +53,7 @@ fn main() {
     let config = toml::from_str(&config).expect("Error parsing config");
 
     vixen::Runtime::builder()
+        .source(YellowstoneGrpcSource::new())
```

Comment:
```suggestion
        .source(YellowstoneGrpcSource)
```

If we don't need wrapper struct `Source::new(YellowstonGrpcSource)` then lets enforce need to expose new and runtime calls it.

Reply:
Hmm, I think we could do something similar but not exactly that, because to pass it like that we need the struct to be a unit struct without data (if I'm not wrong), but what we could do is something like this:
 ```rust
  .source::<YellowstoneGrpcSource>()
  ``` 
 
 (we need to enforce `new()` through another trait though Source can't return owned values cause it's `dyn`, but I tried and works well)